anta/catalog.py

```diff
@@ -94,8 +111,7 @@ def instantiate_inputs(
                 f"{test_class.name} test inputs are not valid: {inputs_msg}\n",
                 {"errors": e.errors()},
             ) from e
-        msg = f"Could not instantiate inputs as type {type(data).__name__} is not valid"
-        raise ValueError(msg)
+        return data
```

Comment:
Hmmm you are changing the output if data is not of correct type - no more meaningful error message?

Reply:
The` raise ValueError` was dead code. If I remove it, pylint complains about returns statements not being consistent, so I'm returning the data here assigned in the try block.
If you end up in the except block, you won't reach this return statement.

Reply:
it is not deadcode if you give data=wrong type as input.

Reply:
Adding a print statement above the return `data`
```
>>> AntaTestDefinition(test = anta.models.AntaTest, inputs=[])
WOUHOU I AM OUT OF TRY EXCEPT RETURNING [] of type <class 'list'>
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/guillaumemulocher/Github/gmuloc/anta/anta/catalog.py", line 70, in __init__
    self.__pydantic_validator__.validate_python(
pydantic_core._pydantic_core.ValidationError: 1 validation error for AntaTestDefinition
inputs
  Input should be a valid dictionary or instance of Input [type=model_type, input_value=[], input_type=list]
    For further information visit https://errors.pydantic.dev/2.6/v/model_type
 ```

Reply:
Ok this was not dead code.
But you got a ValidationError, so what's your point ?

Reply:
It just changes the error message.
The previous one was more or less the same level of info. The default from pydantic indicates the correct type to use:
```
>>> AntaTestDefinition(test = anta.models.AntaTest, inputs=[])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mtache/Git/projects/anta/anta/catalog.py", line 70, in __init__
    self.__pydantic_validator__.validate_python(
pydantic_core._pydantic_core.ValidationError: 1 validation error for AntaTestDefinition
inputs
  Value error, Could not instantiate inputs as type list is not valid [type=value_error, input_value=[], input_type=list]
    For further information visit https://errors.pydantic.dev/2.7/v/value_error
```